woocommerce/woocommerce-android
# Show outdated data

Closes: #11797 
Closes: #11798
Closes: #11791

### Why 
As part of the new Background Tasks projects we aim to improve performance of the app by displaying cached data by default.

### Description
This PR aims to enhance the perceived performance of the dashboard UI by leveraging cached data more effectively.

#### Key Changes

Introduce isOutdated Field:

- Added the isOutdated field to several data classes.
- This field helps differentiate between outdated and up-to-date information.

Refreshing Indicator:

- When outdated information is displayed, a refreshing indicator will inform merchants that the data is being updated in the background.

These changes improve the user experience by ensuring that merchants are aware of data updates while benefiting from faster load times due to caching.

### Testing information
TC1 

1. Open the app and wait for the dashboards cards to load
2. Close the app and wait 30 min for the cached data to expire
3. Open the app and check that the expired data is displayed 

To skip waiting for 30 min you can run this Patch
```
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt	(revision 9f2048dbe3d6cbf00d40b0fb55c3343674154738)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt	(date 1720841339377)
@@ -133,7 +133,7 @@
 
     private fun isElapsedTimeExpired(lastUpdateTime: Long?, maxOutdatedTime: Long): Boolean =
         lastUpdateTime?.let {
-            (currentTime - it) > maxOutdatedTime
+            (currentTime - it) > maxOutdatedTime || true
         } ?: true
 
     private val StatsTimeRangeSelection.identifier

```

TC2
1. Open the app and wait for the dashboards cards to load
2. Pull to refresh the data and check that the stale cached data is displayed while the new data is loading 

### Images/gif

#### Before

Opening the app with stale cache

https://github.com/user-attachments/assets/4030e85f-bfad-4e53-a9d8-e1b5b3527079

Refreshing the Dashboard

https://github.com/user-attachments/assets/1ab01110-7691-47bb-9472-f5500e1ce35a

#### After

Opening the app with stale cache

https://github.com/user-attachments/assets/d0a934ae-b4e0-46de-a123-ff3df0e4a121

Refreshing the Dashboard

https://github.com/user-attachments/assets/809e5304-b6ce-400e-94df-3c6355f5d376
